### PR TITLE
Add ut for prepare.create_collection_request

### PR DESCRIPTION
**Pull request by XuanYang-cn**
Signed-off-by: XuanYang-cn <xuan.yang@zilliz.com>

/hold

#1020

### CODE REVIEW

File: pymilvus/orm/schema.py [hunk 1]
```diff
@@ -10,66 +10,188 @@
...
-                    raise PrimaryKeyException(0, ExceptionsMessage.PrimaryKeyOnlyOne)
-                self._primary_field = field
-                primary_field = field.name
-
-        if self._primary_field is None:
-            raise PrimaryKeyException(0, ExceptionsMessage.PrimaryKeyNotExist)
+class FieldSchema:
+    name: str
+    auto_id: bool = False
+    is_primary: bool = False
```
Comment (longjiquan) [inline pymilvus/orm/schema.py:33]: Why these three static variables should be added? I thought maybe they shouldn't share between `FieldSchema`.

File: pymilvus/client/prepare.py [hunk 2]
```diff
@@ -1,138 +1,142 @@
...
-                #             field_schema.index_params.append(kv_pair)
-
-                schema.fields.append(field_schema)
-
-        consistency_level = get_consistency_level(kwargs.get("consistency_level", DEFAULT_CONSISTENCY_LEVEL))
-        return milvus_types.CreateCollectionRequest(collection_name=collection_name,
-                                                    schema=bytes(schema.SerializeToString()), shards_num=shards_num,
-                                                    consistency_level=consistency_level)
+        for field in all_fields:
+            field_name = field.get('name')
```
[the remark this answers is not in the document]
Comment (XuanYang-cn) [reply pymilvus/client/prepare.py:84]: No, `get` has a default of "None"
```python
Python 3.6.9 (default, Dec  8 2021, 21:08:43)
[GCC 8.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> a = {"x": 1}
>>> a.get("b")
>>> a.get("x")
1
>>>
```

File: pymilvus/client/prepare.py [hunk 3]
```diff
@@ -1,138 +1,142 @@
...
+            if field_name is None:
+                raise ParamError("You should specify the name of field!")
+
+            data_type = field.get('type')
```
[the remark this answers is not in the document]
Comment (XuanYang-cn) [reply pymilvus/client/prepare.py:88]: as above

File: pymilvus/orm/schema.py (hunk 1 above)
Comment (XuanYang-cn) [reply pymilvus/orm/schema.py:33]: This is initialization and a type hint for the non-property fields and it'll be handy for the type checks to be added later.

Also, I'm not fond of the error when we try to get `is_primary` from non-primary field.

```python
>>> class abc:
...     yx: str = ""
...
>>> abc = abc()
>>> abc.yx
''
>>> abc.ljq
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'abc' object has no attribute 'ljq'
>>>
```

**Comment by longjiquan** [comment]
/lgtm
/approve
@XuanYang-cn Great job! thx for the contribution to do the refactor.

**Pull request merged**